plugins/trace_api_plugin/test/test_data_handlers.cpp

```diff
@@ -135,6 +135,52 @@ BOOST_AUTO_TEST_SUITE(abi_data_handler_tests)
       BOOST_TEST(to_kv(expected) == to_kv(std::get<0>(actual)), boost::test_tools::per_element());
    }
 
+   BOOST_AUTO_TEST_CASE(abi_fail_yield)
+   {
+      auto action = action_trace_v1 {
+          { 0, "alice"_n, "alice"_n, "foo"_n, {}, {0x00, 0x01, 0x02, 0x03}},
+          {0x04, 0x05, 0x06, 0x07}
+      };
+
+      std::variant<action_trace_v0, action_trace_v1> action_trace_t = action;
+
+      auto abi = chain::abi_def ( {},
+                                   {
+                                    { "foo", "", { {"a", "varuint32"}, {"b", "varuint32"}, {"c", "varuint32"}, {"d", "varuint32"} } }
+                                   },
+                                   {
+                                    { "foo"_n, "foo", ""}
+                                   },
+                                   {}, {}, {}
+      );
+      abi.version = "eosio::abi/1.";
+
+      bool except_called = false;
+      auto except_handler = [&](const exception_with_context& e) {
+         except_called = true;
+         BOOST_CHECK(std::get<0>(e).operator bool());
+         BOOST_CHECK(std::get<2>(e) > 0);
+         if (std::get<0>(e)) { // rethrow so caller is notified of error
+            std::rethrow_exception(std::get<0>(e));
+         }
+      };
+
+      abi_data_handler handler(except_handler);
+      handler.add_abi("alice"_n, abi);
+
+      fc::variant expected = fc::mutable_variant_object()
+          ("a", 0)
+          ("b", 1)
+          ("c", 2)
+          ("d", 3);
+
+      uint32_t depth = 0;
+      BOOST_CHECK_EXCEPTION(handler.serialize_to_variant(action_trace_t, [&](){ ++depth; if (depth > 1) throw std::runtime_error("oops"); }), std::runtime_error,
+                            [](const std::runtime_error& e) {
+                               return std::string(e.what()).find("oops") != std::string::npos;
+                            });
```

Comment:
Probably you should either check `except_called == true` here, or remove the variable alltogether.

Reply:
PS: cool test. Did you make up the abi out of thin air?

Reply:
Thanks. I meant to add a check for that. Added.
It was copied from another test. Originally hand-crafted.